### ISSUE TITLE
Fix broken badges with statistics

### ISSUE THREAD

**Issue by Aracki** [tldr-pages/tldr#3111]
I suggest we remove the following two badges. 	

* `GitHub contributors` is not needed as you can see that in the top bar.
* `Number of files` seems broken and nevertheless I think we do not need this (maybe the number of pages by OS would be better statistic)

**Comment by waldyrious** [comment]
I agree with removing the number of files badge, Although I'd prefer if we could replace it with another service providing a count of pages in the repo or in a specific folder. I think it was @pxgamer who mentioned that we could even contribute new badges to shields.io, so even if such a badge doesn't exist, maybe we could create it.

As for the contributors badge, I think it's worth keeping it since it matches well with the ones counting merged PRs and issue resolution time. Essentially they're signals of community activity, and the fact that we chose to include those numbers in the README also reflects our project's focus on welcoming contributions.

**Comment by sbrl** [comment]
That's a really good argument for keeping the page count badge, actually.

On Gitter, we considered a few approaches, all centred around the new [dynamic badge](https://shields.io/#dynamic-badge) feature of shields.io:

1. Reconfigure @tldr-bot to emit statistics.
2. Alter the build process to emit a `stats.json` containing various interesting statistics.

Personally, I really like the 2nd option here. It could even be used to make a 'status panel' that shows various things like page counts per platform, translation percentages, etc.

**Comment by Aracki** [comment]
@waldyrious Why there is **406** contributors showed in the badge and **741** in the top bar?

**Comment by pxgamer** [comment]
Anon users maybe? You can do:

**GitHub Contributors (current badge)**
![Contributors](https://img.shields.io/github/contributors/tldr-pages/tldr.svg)
**With anonymous contributions**
![Contributors with Anon](https://img.shields.io/github/contributors-anon/tldr-pages/tldr.svg)

Maybe GitHub has merged some duplicates or something?

**Comment by waldyrious** [comment]
It would indeed be nice to have a more concrete idea of the reason for the discrepancy. That would be relevant e.g. for #1076.

**Comment by mebeim** [comment]
For the number of files badge, the second option proposed by @sbrl seems like the way to go IMHO. I think having some kind of statistics about the project would be cool and helpful, and that would of course make the file count badge trivial to generate from shields.io's dynamic badges.

For the PR badge the thing is, as you guys pointed out, that it doesn't match any other contributor count (apart from maybe being redundant, but as @waldyrious said it's a good indicator of this project's main objective) . If that matched, then I think that'd make more sense, but since it doesn't it actually ends up looking kind of confusing. 

---

PS:  some stats which I think could be cool to have if we decide to produce them:

- Overall page count and page count per OS.
- Translation progress per language.
- Translation progress for coreutils per language (+ details on missing ones).
- Overall size of all pages (maybe even per language?).
- Number of open "page request" issues (slightly more complex than previous ones).

**Comment by waldyrious** [comment]
> Overall size of all pages

Size measured in what? Number of examples? And how is this useful information?

Also, let's make sure we don't overload this plan with too much stuff, otherwise it'll never get off the ground! 😅 I'd say let's just focus on providing a count of pages for now.

**Comment by pxgamer** [comment]
I assumed it was count in MB, KB, or bytes.

But I agree it would be good to get it done initially, we can add more later if necessary.

**Comment by mebeim** [comment]
@waldyrious size in bytes. That'd be a useful indicator of the actual *weight* of the pages on a system. Anyway, I was just throwing out some ideas, since I liked the idea of having some statistics. All of those would be very easily doable along with the page count, but yes that'd still be the most important one.

**Comment by Aracki** [comment]
I think the size in bytes is not useful at all. People would only care about the actual **number** of pages/number per OS.

**Comment by pxgamer** [comment]
@Aracki, I actually kind of agree with @mebeim. With the tldr clients, they download the entire ZIP archive for caching. I'd want to know the size of what I'm downloading. 🤔 (Maybe that's just me though) 😛

The number of pages is also important obviously, but the size isn't a bad stat to include. It's currently 8.3M to store the cache directory in `~/.tldr/cache`

**Comment by mebeim** [comment]
@pxgamer yep that was exactly my thought, but again I was just throwing some ideas 😅

**Comment by sbrl** [comment]
If the stats json file is located alongside the zip archive, then it would make sense to also include the size of the archive file in bytes as a property. If however the stats file is located inside the archive, then it's a bit challenging to include the size of the archive because it won't have been generated yet - and by the time the user has downloaded the archive file they'll know how big it is anyway :P

In other words, if we include the size of the archive as a statistic in the stats file, the stats file should be located alongside the archive so that they can be downloaded independently of each other.

**Comment by pxgamer** [comment]
Alongside the ZIP, definitely not in it. :+1:

**Comment by mebeim** [comment]
Well, of course any stats would be in a separate file, just like index.json is also served as a separate file now, there would be a stats.json file in the same directory, but not inside the ZIP. Point is, anyone who downloads the ZIP already has everything they need to create any kind of statistics they want, so having them inside of it would definitely be redundant.

**Comment by sbrl** [comment]
I'll rename this issue to better reflect what we're going to do then :-)

**Comment by andrik** [comment]
Hi guys, I'm a big fan of translation progress stats even because it was the way I started to contribute.

**Comment by Aracki** [comment]
I would like to close this one if we agreed on keeping the `contributors` badge...?

**Comment by mebeim** [comment]
@Aracki I think we should leave this open until we have implemented a simple statistic as we agreed in the comments above. There's no rush.

**Comment by Managor** [comment]
Does anyone have any interest in this? If not I'll close this.
